rest/model/pod.go

```diff
@@ -70,3 +72,164 @@ func (p *APICreatePod) ToService() (interface{}, error) {
+		Status:                    *s,
+		Secret:                    utility.FromStringPtr(p.Secret),
+		TaskContainerCreationOpts: *taskCreationOpts,
+		Resources:                 resources,
+	}, nil
+}
+
+// APIPodStatus represents a pod's status.
+type APIPodStatus string
+
+const (
+	PodStatusInitializing APIPodStatus = "initializing"
+	PodStatusStarting     APIPodStatus = "starting"
+	PodStatusRunning      APIPodStatus = "running"
+	PodStatusTerminated   APIPodStatus = "terminated"
+)
+
+// BuildFromService converts a service-layer pod status into a REST API pod
+// status.
+func (s *APIPodStatus) BuildFromService(ps pod.Status) error {
+	var converted APIPodStatus
+	switch ps {
+	case pod.StatusInitializing:
+		converted = PodStatusInitializing
+	case pod.StatusStarting:
+		converted = PodStatusStarting
+	case pod.StatusRunning:
+		converted = PodStatusRunning
+	case pod.StatusTerminated:
+		converted = PodStatusTerminated
+	default:
+		return errors.Errorf("unrecognized pod status '%s'", ps)
+	}
+	*s = converted
+	return nil
+}
+
+// ToService converts a REST API pod status into a service-layer pod status.
+func (s *APIPodStatus) ToService() (*pod.Status, error) {
+	if s == nil {
+		return nil, errors.New("invalid pod status")
+	}
+	var converted pod.Status
+	switch *s {
+	case PodStatusInitializing:
+		converted = pod.StatusTerminated
+	case PodStatusStarting:
+		converted = pod.StatusStarting
+	case PodStatusRunning:
+		converted = pod.StatusRunning
+	case PodStatusTerminated:
+		converted = pod.StatusTerminated
+	default:
+		return nil, errors.Errorf("unrecognized pod status '%s'", *s)
+	}
+	return &converted, nil
+}
+
+// APIPodTaskContainerCreationOptions represents options to apply to the task's
+// container when creating a pod.
+type APIPodTaskContainerCreationOptions struct {
+	Image      *string                `json:"image,omitempty"`
+	MemoryMB   *int                   `json:"memory_mb,omitempty"`
+	CPU        *int                   `json:"cpu,omitempty"`
+	Platform   *evergreen.PodPlatform `json:"platform,omitempty"`
+	EnvVars    map[string]string      `json:"env_vars,omitempty"`
+	EnvSecrets map[string]string      `json:"env_secrets,omitempty"`
+}
+
+// BuildFromService converts service-layer task container creation options into
+// REST API task container creation options.
+func (o *APIPodTaskContainerCreationOptions) BuildFromService(opts pod.TaskContainerCreationOptions) {
+	o.Image = utility.ToStringPtr(opts.Image)
+	o.MemoryMB = utility.ToIntPtr(opts.MemoryMB)
+	o.CPU = utility.ToIntPtr(opts.CPU)
+	o.Platform = &opts.Platform
+	o.EnvVars = opts.EnvVars
+	o.EnvSecrets = opts.EnvSecrets
+}
+
+// ToService converts REST API task container creation options into
+// service-layer task container creation options.
+func (i *APIPodTaskContainerCreationOptions) ToService() (*pod.TaskContainerCreationOptions, error) {
+	if i.Platform == nil {
+		return nil, errors.New("missing platform")
```

Comment:
maybe instead this case should set platform to ""?

Reply:
The empty string is still an invalid platform. I assume the calling code won't continue if this returns an error.